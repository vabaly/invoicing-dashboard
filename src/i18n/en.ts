const en = {
  siteInfo: {
    title: 'Attelas invoicing dashboard',
    description: 'Attelas invoicing dashboard',
  },

  breadcrumbs: {
    home: 'Invoices',
  },

  invoiceTableHead: {
    invoiceNumber: 'Invoice Number',
    clientName: 'Client Name',
    total: 'Total Amount Due',
    dueDate: 'Due Date',
    status: 'Status',
  },

  invoiceDetail: {
    basicInfo: 'Basic Info',
    clientInfo: 'Client Info',
    lineItems: 'Line Items',

    email: 'Email',
    address: 'Address',
    clientId: 'Client ID',
  },

  lineItem: {
    description: 'Description',
    price: 'Price',
    quantity: 'Quantity',
  },

  back: 'Back',
  ok: 'Ok',
  colon: ': ',

  chase: 'Chase',

  email: {
    title: 'Email preview',
    to: 'To',
    subject: 'Subject',
    subjectValue: 'Please check your invoice',
    bodies: {
      0: 'Dear %{name},',
      1: 'This is your invoice details. Please pay it as soon as you can.',
      2: 'Best regards,',
      3: 'Attelas Team',
    },
  },
};

export default en;
