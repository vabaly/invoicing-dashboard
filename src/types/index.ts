import { type InvoiceWithClient } from '~/business';

export interface InvoiceProps {
  invoice: InvoiceWithClient;
}
