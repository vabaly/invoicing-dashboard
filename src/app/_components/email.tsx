import { Chip, Divider, Typography } from '@mui/material';
import i18n from '~/i18n';
import { type InvoiceProps } from '~/types';
import { InvoiceItem } from './invoice';
import { addCommasToNumber } from '~/utils';
// import { InvoiceDetail } from './invoice';

export function FakeEmail({ invoice }: InvoiceProps) {
  const { client } = invoice;

  if (!client) {
    return null;
  }

  return (
    <div className="w-full border px-4 md:w-[500px]">
      <div className="flex items-center space-x-2 border-b p-4">
        <Typography className="min-w-14 text-gray-500" variant="body2">
          {i18n.t('email.to')}
        </Typography>
        <Chip label={client.email} size="small" />
      </div>
      <div className="flex items-center space-x-2 border-b p-4">
        <Typography className="min-w-14 text-gray-500" variant="body2">
          {i18n.t('email.subject')}
        </Typography>
        <Typography variant="body2">{i18n.t('email.subjectValue')}</Typography>
      </div>
      <div className="max-h-[350px] space-y-2 overflow-auto p-4 md:max-h-[450px]">
        <Typography variant="body1">
          {i18n.t('email.bodies.0', { name: invoice.client?.name ?? 'user' })}
        </Typography>
        <Typography className="pb-2" variant="body1">
          {i18n.t('email.bodies.1')}
        </Typography>
        <Divider />
        {/* <InvoiceDetail small invoice={invoice} /> */}
        <InvoiceItem
          small
          title={i18n.t('invoiceTableHead.invoiceNumber')}
          text={invoice.invoiceNumber}
        />
        <InvoiceItem
          small
          title={i18n.t('invoiceTableHead.total')}
          text={addCommasToNumber(invoice.total)}
        />
        <InvoiceItem
          small
          title={i18n.t('invoiceTableHead.dueDate')}
          text={invoice.dueDate}
        />
        <Divider />
        <Typography className="pt-2" variant="body1">
          {i18n.t('email.bodies.2')}
        </Typography>
        <Typography variant="body1">{i18n.t('email.bodies.3')}</Typography>
      </div>
    </div>
  );
}
