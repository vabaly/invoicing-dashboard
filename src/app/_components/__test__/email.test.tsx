import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { FakeEmail } from '../email';
import i18n from '~/i18n';
import { type InvoiceProps } from '~/types';

describe('FakeEmail', () => {
  let invoiceProps: InvoiceProps;

  beforeEach(() => {
    invoiceProps = {
      invoice: {
        clientId: 'test',
        client: {
          email: 'test@example.com',
          name: 'Test User',
          clientId: 'test',
          address: '123 Main St',
        },
        invoiceNumber: '12345',
        total: 100,
        dueDate: 'test',
        status: 'pending',
        lineItems: [],
      },
    };
  });

  test('should render when client is present', () => {
    const { asFragment } = render(<FakeEmail {...invoiceProps} />);

    // Snapshot test
    expect(asFragment().children[0]).toMatchSnapshot();

    expect(screen.getByText(i18n.t('email.to'))).toBeInTheDocument();
    if (invoiceProps.invoice.client?.email) {
      const texts = screen.getAllByText(invoiceProps.invoice.client.email);
      expect(texts.length).toBe(2);
      // Pick one to valid
      expect(texts[0]).toBeInTheDocument();
    }
    expect(screen.getByText(i18n.t('email.subject'))).toBeInTheDocument();
    expect(screen.getByText(i18n.t('email.subjectValue'))).toBeInTheDocument();
    expect(
      screen.getByText(
        i18n.t('email.bodies.0', {
          name: invoiceProps.invoice.client?.name ?? 'user',
        }),
      ),
    ).toBeInTheDocument();
    expect(screen.getByText(i18n.t('email.bodies.1'))).toBeInTheDocument();
    expect(screen.getByText(i18n.t('email.bodies.2'))).toBeInTheDocument();
    expect(screen.getByText(i18n.t('email.bodies.3'))).toBeInTheDocument();
  });

  test('should not render when client is not present', () => {
    if (invoiceProps.invoice.client) {
      invoiceProps.invoice.client = undefined;

      render(<FakeEmail {...invoiceProps} />);

      expect(screen.queryByText(i18n.t('email.to'))).not.toBeInTheDocument();
      expect(
        screen.queryByText(i18n.t('email.subject')),
      ).not.toBeInTheDocument();
      expect(
        screen.queryByText(i18n.t('email.subjectValue')),
      ).not.toBeInTheDocument();
      expect(
        screen.queryByText(i18n.t('email.bodies.1')),
      ).not.toBeInTheDocument();
      expect(
        screen.queryByText(i18n.t('email.bodies.2')),
      ).not.toBeInTheDocument();
      expect(
        screen.queryByText(i18n.t('email.bodies.3')),
      ).not.toBeInTheDocument();
    }
  });
});
