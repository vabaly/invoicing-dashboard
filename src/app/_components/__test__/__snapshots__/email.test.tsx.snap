// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FakeEmail should render when client is present 1`] = `
<div
  class="w-full border px-4 md:w-[500px]"
>
  <div
    class="flex items-center space-x-2 border-b p-4"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 min-w-14 text-gray-500 css-1nmqws6-MuiTypography-root"
    >
      To
    </p>
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-14tabw5-MuiChip-root"
    >
      <span
        class="MuiChip-label MuiChip-labelSmall css-3fh3fk-MuiChip-label"
      >
        test@example.com
      </span>
    </div>
  </div>
  <div
    class="flex items-center space-x-2 border-b p-4"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 min-w-14 text-gray-500 css-1nmqws6-MuiTypography-root"
    >
      Subject
    </p>
    <p
      class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
    >
      Please check your invoice
    </p>
  </div>
  <div
    class="max-h-[350px] space-y-2 overflow-auto p-4 md:max-h-[450px]"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
    >
      Dear Test User,
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 pb-2 css-1edfpdg-MuiTypography-root"
    >
      This is your invoice details. Please check it.
    </p>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="space-y-4"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 md:pl-4 css-8zgow4-MuiTypography-root"
      >
        Basic Info
      </h6>
      <div
        class="grid grid-cols-1 gap-4"
      >
        <div
          class="md:flex md:items-center"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 text-gray-500 md:min-w-40 md:text-right css-1ixrkmk-MuiTypography-root"
          >
            Invoice Number: 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 md:pl-8 css-1edfpdg-MuiTypography-root"
          >
            12345
          </p>
        </div>
        <div
          class="md:flex md:items-center"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 text-gray-500 md:min-w-40 md:text-right css-1ixrkmk-MuiTypography-root"
          >
            Due Date: 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 md:pl-8 css-1edfpdg-MuiTypography-root"
          >
            test
          </p>
        </div>
        <div
          class="md:flex md:items-center"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 text-gray-500 md:min-w-40 md:text-right css-1ixrkmk-MuiTypography-root"
          >
            Total Amount Due: 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 md:pl-8 css-1edfpdg-MuiTypography-root"
          >
            100
          </p>
        </div>
        <div
          class="md:flex md:items-center"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 text-gray-500 md:min-w-40 md:text-right css-1ixrkmk-MuiTypography-root"
          >
            Status: 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 md:pl-8 css-1edfpdg-MuiTypography-root"
          >
            pending
          </p>
        </div>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6 pt-4 md:pl-4 css-8zgow4-MuiTypography-root"
      >
        Client Info
      </h6>
      <div
        class="grid grid-cols-1 gap-4"
      >
        <div
          class="md:flex md:items-center"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 text-gray-500 md:min-w-40 md:text-right css-1ixrkmk-MuiTypography-root"
          >
            Client ID: 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 md:pl-8 css-1edfpdg-MuiTypography-root"
          >
            test
          </p>
        </div>
        <div
          class="md:flex md:items-center"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 text-gray-500 md:min-w-40 md:text-right css-1ixrkmk-MuiTypography-root"
          >
            Client Name: 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 md:pl-8 css-1edfpdg-MuiTypography-root"
          >
            Test User
          </p>
        </div>
        <div
          class="md:flex md:items-center"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 text-gray-500 md:min-w-40 md:text-right css-1ixrkmk-MuiTypography-root"
          >
            Email: 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 md:pl-8 css-1edfpdg-MuiTypography-root"
          >
            test@example.com
          </p>
        </div>
        <div
          class="md:flex md:items-center"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 text-gray-500 md:min-w-40 md:text-right css-1ixrkmk-MuiTypography-root"
          >
            Address: 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 md:pl-8 css-1edfpdg-MuiTypography-root"
          >
            123 Main St
          </p>
        </div>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6 pt-4 md:pl-4 css-8zgow4-MuiTypography-root"
      >
        Line Items
      </h6>
      <table
        class="MuiTable-root max-w-[500px] css-rqglhn-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1t7lzbz-MuiTableCell-root"
              scope="col"
            >
              Description
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-g2aoha-MuiTableCell-root"
              scope="col"
            >
              Price
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-g2aoha-MuiTableCell-root"
              scope="col"
            >
              Quantity
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        />
      </table>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 pt-2 css-1edfpdg-MuiTypography-root"
    >
      Best regards,
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
    >
      Attelas Team
    </p>
  </div>
</div>
`;
